### PR TITLE
[ODS-5908] Update context based routing to work with uniform base route

### PR DESCRIPTION
**Pull request by gmcelhanon**

### CODE REVIEW

File: Application/EdFi.Ods.Common/Configuration/ApiSettings.cs [hunk 1]
```diff
@@ -6,14 +6,18 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.RegularExpressions;
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Configuration/ApiSettings.cs:9]: This reference is not being used, can be removed

File: Application/EdFi.Ods.Common/Configuration/ApiSettings.cs [hunk 2]
```diff
@@ -6,14 +6,18 @@
...
 using EdFi.Common.Configuration;
 using EdFi.Common.Extensions;
+using EdFi.Ods.Common.Extensions;
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Common/Configuration/ApiSettings.cs:12]: This reference is not being used, can be removed

**Review by simpat-jesus — CHANGES_REQUESTED** [review]
Looks good! some minor comments, once fixed this can be merged

File: Application/EdFi.Ods.Api/Conventions/IOdsRouteRootTemplateProvider.cs [hunk 3]
```diff
@@ -10,12 +10,11 @@ namespace EdFi.Ods.Api.Conventions;
 /// <summary>
 /// Defines a method for obtaining the root route template prefix for controllers using the specified <see cref="RouteContextType" />.
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Conventions/IOdsRouteRootTemplateProvider.cs]: Done.

File: Application/EdFi.Ods.Api/Conventions/OdsRouteRootTemplateProvider.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+using EdFi.Common.Extensions;
+using EdFi.Ods.Api.Attributes;
+using EdFi.Ods.Api.Constants;
+using EdFi.Ods.Common.Configuration;
+using EdFi.Ods.Common.Constants;
+
+namespace EdFi.Ods.Api.Conventions;
+
+/// <summary>
+/// Obtains the root route template prefix for controllers using the specified <see cref="RouteContextType" />.
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Conventions/OdsRouteRootTemplateProvider.cs]: Done.

File: Application/EdFi.Ods.Api/Conventions/OdsRouteRootTemplateProvider.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.Common.Extensions;
+using EdFi.Ods.Api.Attributes;
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Conventions/OdsRouteRootTemplateProvider.cs]: Done.

File: Application/EdFi.Ods.Common/Configuration/ApiSettings.cs (hunk 1 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Configuration/ApiSettings.cs]: Done.

File: Application/EdFi.Ods.Common/Configuration/ApiSettings.cs (hunk 2 above)
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Configuration/ApiSettings.cs]: Done.

**Review by simpat-jesus — APPROVED** [review]

**Pull request merged**